src/app/shared/details/details.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'ngx-details-access',
+  templateUrl: './details.component.html',
+  styleUrls: ['./details.component.scss'],
+})
+export class DetailsComponent {
+  // Depends on valuePrepareFunction
+  value: any;
+  title: any;
+
+  constructor(
+    private router: Router,
+  ) { }
+
+  onOpenDetailsPage() {
+    let route = '';
+
+    if (this.value.type === 'gateway') {
```

Comment:
Can you use object literal here ? [Here is a example](https://ultimatecourses.com/blog/deprecating-the-switch-statement-for-object-literals)